### ISSUE TITLE
🐛 Bug Report: plugin-kubernetes-backend has incorrect config schema

### ISSUE THREAD

**Issue by rmartine-ias** [backstage/backstage#16837]
### 📜 Description

I think the [config](https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/config.d.ts) that `backstage-cli config:check` uses to validate `app-config.yaml` has an outdated list of `objectTypes`; it does not include `limitranges`, `statefulsets`, and `daemonsets`, specified [here](https://github.com/backstage/backstage/blob/80698c1de81dc0774b87c466cb8951c11d4acea4/plugins/kubernetes-backend/src/types/types.ts#L93).

### 👍 Expected behavior

Checking the config when those types are present succeeds.

### 👎 Actual Behavior with Screenshots

```shell
$ backstage-cli config:check --config app-config.yaml --lax
Loaded config from app-config.yaml

Error: Configuration does not match schema

  Config must be equal to one of the allowed values { allowedValues=configmaps,cronjobs,deployments,horizontalpodautoscalers,ingresses,jobs,pods,replicasets,services } at /kubernetes/objectTypes/8
  Config must be equal to one of the allowed values { allowedValues=configmaps,cronjobs,deployments,horizontalpodautoscalers,ingresses,jobs,pods,replicasets,services } at /kubernetes/objectTypes/9
  Config must be equal to one of the allowed values { allowedValues=configmaps,cronjobs,deployments,horizontalpodautoscalers,ingresses,jobs,pods,replicasets,services } at /kubernetes/objectTypes/10
```

With `app-config.yaml` containing:

```yaml
kubernetes:
  objectTypes:
    - pods
    - services
    - configmaps
    - deployments
    - replicasets
    - horizontalpodautoscalers
    - jobs
    - ingresses
    - limitranges
    - statefulsets
    - daemonsets
```

### 👟 Reproduction steps

Add above yaml to a new app-config.yaml, install kubernetes plugin, check config.

### 📃 Provide the context for the Bug.

Since https://github.com/backstage/backstage/issues/6109 is an issue for us (using k8s 1.19), we need to explicitly specify the `objectTypes`, sans `cronjobs`. If we write out all `objectTypes` except `cronjobs`, `backstage-cli config:check` fails.

### 🖥️ Your Environment

```shell
yarn run v1.22.19
$ /Users/rmartine/dev/ias-backstage/node_modules/.bin/backstage-cli info
OS:   Darwin 22.3.0 - darwin/arm64
node: v16.18.1
yarn: 1.22.19
cli:  0.22.3 (installed)
backstage:  1.8.0

Dependencies:
  @backstage/app-defaults                          1.1.0
  @backstage/backend-app-api                       0.4.0
  @backstage/backend-common                        0.18.2
  @backstage/backend-dev-utils                     0.1.0
  @backstage/backend-plugin-api                    0.4.0
  @backstage/backend-tasks                         0.4.3
  @backstage/catalog-client                        1.3.1
  @backstage/catalog-model                         1.2.0
  @backstage/cli-common                            0.1.11
  @backstage/cli                                   0.22.3
  @backstage/config-loader                         1.1.8
  @backstage/config                                1.0.6
  @backstage/core-app-api                          1.4.0
  @backstage/core-components                       0.12.4
  @backstage/core-plugin-api                       1.4.0
  @backstage/errors                                1.1.4
  @backstage/eslint-plugin                         0.1.1
  @backstage/integration-aws-node                  0.1.1
  @backstage/integration-react                     1.1.10
  @backstage/integration                           1.4.2
  @backstage/plugin-api-docs                       0.8.14
  @backstage/plugin-app-backend                    0.3.42
  @backstage/plugin-auth-backend                   0.18.0
  @backstage/plugin-auth-node                      0.2.11
  @backstage/plugin-catalog-backend-module-github  0.2.5
  @backstage/plugin-catalog-backend                1.7.2
  @backstage/plugin-catalog-common                 1.0.11
  @backstage/plugin-catalog-graph                  0.2.26
  @backstage/plugin-catalog-import                 0.9.4
  @backstage/plugin-catalog-node                   1.3.3
  @backstage/plugin-catalog-react                  1.3.0
  @backstage/plugin-catalog                        1.7.2
  @backstage/plugin-events-node                    0.2.3
  @backstage/plugin-github-actions                 0.5.14
  @backstage/plugin-kubernetes-backend             0.9.3
  @backstage/plugin-kubernetes-common              0.6.0
  @backstage/plugin-kubernetes                     0.7.8
  @backstage/plugin-org                            0.6.4
  @backstage/plugin-permission-backend             0.5.17
  @backstage/plugin-permission-common              0.7.3
  @backstage/plugin-permission-node                0.7.5
  @backstage/plugin-permission-react               0.4.10
  @backstage/plugin-proxy-backend                  0.2.35
  @backstage/plugin-scaffolder-backend             1.11.0
  @backstage/plugin-scaffolder-common              1.2.5
  @backstage/plugin-scaffolder-node                0.1.0
  @backstage/plugin-scaffolder-react               1.1.0
  @backstage/plugin-scaffolder                     1.11.0
  @backstage/plugin-search-backend-module-pg       0.5.3
  @backstage/plugin-search-backend-node            1.1.3
  @backstage/plugin-search-backend                 1.2.2
  @backstage/plugin-search-common                  1.2.1
  @backstage/plugin-search-react                   1.4.0
  @backstage/plugin-search                         1.0.7
  @backstage/plugin-tech-radar                     0.5.20
  @backstage/plugin-techdocs-backend               1.5.2
  @backstage/plugin-techdocs-module-addons-contrib 1.0.9
  @backstage/plugin-techdocs-node                  1.4.5
  @backstage/plugin-techdocs-react                 1.1.2
  @backstage/plugin-techdocs                       1.4.3
  @backstage/plugin-user-settings                  0.5.1
  @backstage/release-manifests                     0.0.8
  @backstage/test-utils                            1.2.4
  @backstage/theme                                 0.2.17
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3
Done in 0.82s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by freben** [comment]
Oh for sure, thanks for reporting and we'd appreciate a PR to add those properly of course!

**Comment by jamieklassen** [comment]
I can recall at least two occasions (#13645,#15349) where a very similar problem has arisen -- the kubernetes-backend plugin starts reading some new config structure which is not reflected in the schema, so configs which are valid from the perspective of the plugin do not pass the `config:check` command.

I wondered if we might be able to prevent this with some automation, and while I was working on #13685 I had an idea for exercising the config schema in integration tests. I ended up hacking around with replacing test setups like

https://github.com/backstage/backstage/blob/764c7facc495e2146d8b749f504c051f99afafbe/plugins/kubernetes-backend/src/service/KubernetesBuilder.test.ts#L47-L52

which create a [`ConfigReader`](https://backstage.io/docs/reference/config.configreader._constructor_/) directly, with ones that hacked up mock-fs and called [`loadBackendConfig`](https://backstage.io/docs/reference/backend-common.loadbackendconfig/#__docusaurus) which actually reads config schemas from disk at some point. However, when I finally got it running I found that providing invalid config wasn't failing any tests! This is because `loadBackendConfig` doesn't validate that the provided config matches the schema. Too bad! But this might be a nice behaviour for Backstage to have, like a "strict config mode" or something.

Anyway, if we wanted to prevent issues like this via integration test utilities, we might need to write something new that explicitly checks the schema -- as far as I can tell, there's nothing prefabricated for this purpose.
